api/src/main/java/edu/cornell/mannlib/vitro/webapp/i18n/I18n.java

```diff
@@ -153,6 +223,28 @@ private void checkForChangeInThemeDirectory(HttpServletRequest req) {
 		}
 	}
 
+	/**
+	 * If the theme directory has changed from before, clear the cache of all
+	 * ResourceBundles.
+	 */
+	private void checkForChangeInThemeDirectory(ServletContext ctx) {
+		WebappDaoFactory wdf = ModelAccess.on(ctx)
```

Comment:
`ctx` may be null, no?

Reply:
I certainly hope not. The `ServletContext` is defined when running in a container and `ServletContextStub` is defined when running tests.

Reply:
The `ServletContext` is final and would always be defined when the instance is instantiated via `setup`.

Reply:
Since `setup` is called as the start-up of a ServletContextListener (LocaleSelectionSetup), we are probably safe. Maybe a comment is sufficient to highlight the dependency on `setup` needing to be called before using this class.

Reply:
I will add a comment. I would prefer another way, but I am failing to see other strategies with less egregious hacks.